iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -116,6 +116,13 @@ private static TableScan createTableScan(Table table, Configuration conf) {
       scan = scan.useSnapshot(snapshotId);
     }
 
+    if (snapshotId == -1) {
```

Comment:
should we put it in `else` block?

Reply:
SNAPSHOT_ID_INTERVAL_FROM handling is moved to else block.
AS_OF_TIMESTAMP was independently set from SNAPSHOT_ID. Is it possible to construct a query where both are present?

Reply:
I don't think so, that would be weird